src/menu.c

```diff
@@ -20,31 +20,41 @@
 #include "settings.h"
 #include "utils.h"
 
+#define ERROR_BUF_LEN 120
+
 static bool is_initialized = false;
 static regex_t cregex;
+static char ERROR_BUF[ERROR_BUF_LEN];
 
 struct notification_lock {
         struct notification *n;
         gint64 timeout;
 };
 static gpointer context_menu_thread(gpointer data);
 
-static int regex_init(void)
+/*
+ * Initialize regexes needed for matching.
+ *
+ * Return: true if initialization succeeded
+ *
+ */
+static bool regex_init(void)
 {
         if (is_initialized)
-                return 1;
+                return true;
 
         char *regex =
             "\\b(https?://|ftps?://|news://|mailto:|file://|www\\.)"
             "[-[:alnum:]_\\@;/?:&=%$.+!*\x27,~#]*"
             "(\\([-[:alnum:]_\\@;/?:&=%$.+!*\x27,~#]*\\)|[-[:alnum:]_\\@;/?:&=%$+*~])+";
-        int ret = regcomp(&cregex, regex, REG_EXTENDED | REG_ICASE);
-        if (ret != 0) {
-                LOG_W("Failed to compile regex.");
-                return 0;
+        int code = regcomp(&cregex, regex, REG_EXTENDED | REG_ICASE);
+        if (code != 0) {
+                regerror(code, &cregex, ERROR_BUF, ERROR_BUF_LEN);
+                LOG_W("Failed to compile regex: %s", ERROR_BUF);
```

Comment:
While reading this code, isn't here an additional `is_initialized = false;` missing?

Reply:
I don't think so - `is_initialized` should already be false, do you want me to do it again to be explicit?

Reply:
Sorry for this pointless suggestion. :see_no_evil: of course we bail out right at the beginning of the method if it's already true, so it has to be false already.